nkms/characters.py

```diff
@@ -226,6 +231,25 @@ def publish_treasure_map(self, policy_group):
         setter = self.server.set(dht_key, b"trmap" + dht_value)
         return setter, encrypted_treasure_map, dht_value, signature_for_bob, signature_for_ursula
 
+    def grant(self, bob, uri, networky_stuff, m=None, n=None, expiration=None, deposit=NON_PAYMENT):
+        if not m:
+            # TODO: get m from config
+            raise NotImplementedError
+        if not n:
+            # TODO: get n from config
+            raise NotImplementedError
+        if not expiration:
+            # TODO: check default duration in config
+            raise NotImplementedError
+
```

Comment:
I'd also take deposit proportional to time from config (e.g. mETH/day/n or something) if deposit is not set. Wdyt?

Reply:
I would research the costs of Amazon KMS and Google KMS and build a competitive algorithm to determine cost/day.

Reply:
Eh, I suspect amazon kms might be way too expensive. We might (?) have some price discovery. But that's not an easy problem.
Still, good to know what they offer

Reply:
I meant to say that we build a pricing algorithm to beat their pricing model. They charge per key, IIRC.